### PR TITLE
fix: Flake8-BugBear code B-026 for PyTorch

### LINKED ISSUE #106571: Enable more flake8-bugbear lints
### 🚀 The feature, motivation and pitch

* There are a lot of flake8-bugbear code reported by flake8-bugbear and ruff that need to be fixed. These would be a good starting issue for a new contributor.

This can be found be removing the codes from the ignore list in the .flake8 file and the .pyproject file for ruff.

- [ ] "B007", 
- [ ] "B008" 
- [ ] "B017",
- [ ] "B018", # Useless expression
- [ ] "B019", 
- [ ] "B020",
- [ ] "B023", 
- [ ] "B024", 
- [ ] "B026",
- [ ] "B028", # No explicit `stacklevel` keyword argument found
- [ ] "B904",

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by guptaaryan16**
Fixes #106571

I have fixed the B-026 error codes for Flake8 tests on the codebase. Please review and tell me anything else to do.
Thanks and excited for this first contribution to PyTorch.

Also I refer this issue which introduced [B-026](https://github.com/PyCQA/flake8-bugbear/issues/286) in `pytest-bugbear` and discuss the error code.

### CODE REVIEW

**Comment by janeyx99** [comment]
Looks like there are needed changes to pass lint in CI!

**Comment by janeyx99** [comment]
@Skylion007 assigning you as reviewer

File: torch/utils/data/datapipes/iter/combinatorics.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def __init__(self,
         self.sampler_args = () if sampler_args is None else sampler_args
         self.sampler_kwargs = {} if sampler_kwargs is None else sampler_kwargs
         # https://github.com/python/mypy/pull/9629 will solve
-        self.sampler = sampler(data_source=self.datapipe, *self.sampler_args, **self.sampler_kwargs)  # type: ignore[misc]
+        self.sampler = sampler(*self.sampler_args, **self.sampler_kwargs, data_source=self.datapipe)  # type: ignore[misc]
```
Comment (Skylion007) [inline torch/utils/data/datapipes/iter/combinatorics.py:42]: Do we expect the user to be able to override the data_source with kwargs?

File: torch/_functorch/aot_autograd.py [hunk 2]
```diff
@@ -4541,7 +4541,7 @@ def functional_call(named_params, named_buffers, *args, **kwargs):
     named_buffers = dict(mod.named_buffers(remove_duplicate=False))
     num_params_buffers = len(named_params) + len(named_buffers)
     compiled_f = aot_function(
-        functional_call, num_params_buffers=num_params_buffers, *args, **kwargs
+        functional_call, *args, **kwargs, num_params_buffers=num_params_buffers
```
Comment (Skylion007) [inline torch/_functorch/aot_autograd.py:4544]: Likewise, do we expect users to be able to override this?

Comment (Skylion007) [reply torch/_functorch/aot_autograd.py:4544]: Maybe we should add an assert that the specific keys is not kwargs.

File: torch/utils/data/datapipes/iter/combinatorics.py (hunk 1 above)
Comment (guptaaryan16) [reply torch/utils/data/datapipes/iter/combinatorics.py:42]: @Skylion007 I checked the rules and seems like we can put kwargs after the `data_source`

```
- self.sampler = sampler(data_source=self.datapipe, *self.sampler_args, **self.sampler_kwargs)  # type: ignore[misc]
+ self.sampler = sampler(*self.sampler_args, data_source=self.datapipe, **self.sampler_kwargs)  # type: ignore[misc]
```
So I will change this for all the file changes. Also should we check args for the overrides?

File: torch/_functorch/aot_autograd.py (hunk 2 above)
Comment (guptaaryan16) [reply torch/_functorch/aot_autograd.py:4544]: Same as above

File: torch/utils/data/datapipes/iter/combinatorics.py (hunk 1 above)
Comment (Skylion007) [reply torch/utils/data/datapipes/iter/combinatorics.py:42]: I'f feel a bit more comfortable with this as it's less likely to be a behavior change.

Comment (guptaaryan16) [reply torch/utils/data/datapipes/iter/combinatorics.py:42]: Sure so lets add an assert for the `data_source` and other arguments in kwargs

Comment (guptaaryan16) [reply torch/utils/data/datapipes/iter/combinatorics.py:42]: Sure so let me make the changes

**Comment by guptaaryan16** [comment]
Hey @Skylion007 can you give another review to the PR?

**Review by Skylion007 — APPROVED** [review]

**Comment by guptaaryan16** [comment]
Hey @Skylion007 thanks for the approval. Seems like I run into a lint error again, can you approve the workflows once again, it will work this time for sure. Thanks

**Comment by guptaaryan16** [comment]
Hey @Skylion007 @janeyx99 can you please approve the workflows so that I can continue with the merge. Thanks

**Review by Skylion007 — APPROVED** [review]

**Comment by guptaaryan16** [comment]
Hey can someone approve the dev workflows for this PR?
cc @Skylion007 @janeyx99

**Pull request closed without merging**